client/visualizer/src/game/gamearea/gamearea.ts

```diff
@@ -125,17 +127,22 @@ export default class GameArea {
       // Reset change listeners
       window.onresize = function() {};
     } else {
-      if (mode === Mode.MAPEDITOR) {
-        this.wrapper.appendChild(this.mapEditorCanvas);
-      } else {
-        this.wrapper.appendChild(this.canvas); // TODO: Only append if a game is available in client.games
-        console.log("Now a game");
+      switch (mode) {
+        case Mode.MAPEDITOR:
+          this.wrapper.appendChild(this.mapEditorCanvas);
+          break;
+        case Mode.PROFILER:
+          this.wrapper.appendChild(this.profilerIFrame);
+          break;
+        default:
+          this.wrapper.appendChild(this.canvas); // TODO: Only append if a game is available in client.games
+          console.log("Now a game");
       }
     }
     
     if(shouldListen) {
       window.onresize = function() {
-        var wrapper = <HTMLDivElement> document.getElementById("canvas-wrapper")
+        var wrapper = <HTMLDivElement> document.getElementById("canvas-wrapper");
```

Comment:
:0

Reply:
I use IntelliJ and it highlighted all the missing semicolons, so I added some as I read over them and eventually gave up because they are used wayyyy too inconsistent :P